airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -538,6 +540,9 @@ def cast_property_type(self, property_name: str, column_name: str, jinja_column:
                 trimmed_column_name = f"trim(BOTH '\"' from {column_name})"
                 sql_type = f"'{sql_type}'"
                 return f"nullif(accurateCastOrNull({trimmed_column_name}, {sql_type}), 'null') as {column_name}"
+            elif self.destination_type == DestinationType.MYSQL:
+                # Cast to `text` datatype. See https://github.com/airbytehq/airbyte/issues/7994
+                sql_type = f"{sql_type}(1024)"
```

Comment:
The proper way of changing a datatype for a certain destination should be done in dbt macros, not directly in the python code:

see https://github.com/airbytehq/airbyte/blob/c7b8db702e8c0505f624101905ef778a12ae10f0/airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/datatypes.sql#L43

Reply:
Could this be causing this error?

Failure Origin: normalization, Message: Normalization failed during the dbt run. This may indicate a problem with the data itself.

  1292 (22007): Truncated incorrect CHAR(1024) value: